stat.c

```diff
@@ -3087,17 +3091,30 @@ int calc_log_samples(void)
 		if (!td->bw_log ||
 			(td->bw_log && !per_unit_log(td->bw_log))) {
 			tmp = add_bw_samples(td, &now);
-			if (tmp < next)
-				next = tmp;
+
+			if (td->bw_log)
+				log_avg_msec = td->bw_log->avg_msec;
 		}
 		if (!td->iops_log ||
 			(td->iops_log && !per_unit_log(td->iops_log))) {
 			tmp = add_iops_samples(td, &now);
-			if (tmp < next)
-				next = tmp;
+
+			if (td->iops_log)
+				log_avg_msec = td->iops_log->avg_msec;
```

Comment:
If both are on it doesn't matter who (iops/bw_log) sets `log_avg_msec` (because they both derive their value from the same original option). However the value MAY only be in one of  `bw_log` or `iops_log` structs so we are actually forced to check both to be safe...

Reply:
yes, that was my thought on this.